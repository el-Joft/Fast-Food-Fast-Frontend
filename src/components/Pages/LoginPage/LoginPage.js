/* eslint-disable react/no-unescaped-entities */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { loginUser } from '../../../actions/authActions';

import './LoginPage.scss';

class LoginPage extends Component {
  state = {
    email: '',
    password: '',
    errors: {},
  }

  componentDidMount () {
    if (this.props.auth.isAuthenticated) {
      this.props.history.push('/');
    }
  }

  componentWillReceiveProps(nextProps) {
    console.log(nextProps);
    if (nextProps.auth.isAuthenticated) {
      nextProps.history.push('/');
    }

    if (nextProps.errors) {
      this.setState({
        errors: nextProps.errors,
      });
    }
  }

  onChange = (event) => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  }

  onSubmit = (event) => {
    event.preventDefault();
    const loginDetails = {
      email: this.state.email,
      password: this.state.password,
    };
    this.props.loginUser(loginDetails);
  };

  render() {
    return (
      <section>
        { this.state.errors === ''
          ? <div className='message' />
          : <div className='error_message'>{this.state.errors.message}</div>
      }
        <div>
          <form onSubmit={this.onSubmit}>
            <ul className="form-group">

              <li>
                <label>
Email
                  {' '}
                  <span className="required">*</span>
                </label>
                <input onChange={this.onChange} value={this.state.email} type="email" name="email" className="form-control field-long" required />
              </li>
              <li>
                <label>
Password
                  {' '}
                  <span className="required">*</span>
                </label>
                <input onChange={this.onChange} value={this.state.password} type="password" name="password" className="form-control field-long" required />
              </li>

              <li>
                <input type="submit" value="Submit" />
              </li>

            </ul>

          </form>
          <div className="text-form">
            <p>
Don't have an account?
              {' '}
              <Link to="/signupl">Register</Link>
            </p>
          </div>
        </div>

      </section>
    );
  }
}

LoginPage.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  // history: PropTypes.func.isRequired,
};

// we get values here using the this.props.errors/auth

const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors,
});


export default connect(mapStateToProps, { loginUser })(LoginPage);
