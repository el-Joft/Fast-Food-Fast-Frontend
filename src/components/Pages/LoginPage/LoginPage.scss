.error_message {
  color: red;
  text-align: center;
  margin-top: 20px;
  padding: 10pxx;
  transition: opacity 1s ease-in;
  display: block;
  opacity: 1;
  height: auto;
}