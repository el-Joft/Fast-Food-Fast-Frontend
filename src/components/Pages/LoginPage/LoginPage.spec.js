/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { LoginPage } from './LoginPage';

describe('rendering', () => {
  let wrapper;
  let props;
  let nextProps;
  beforeEach(() => {
    props = {
      loginUser: jest.fn(),
      checkoutOrder: jest.fn(),
      history: {
        push: jest.fn(),
      },
      success: {
        status: true,
      },
      orders: [],
      auth: {
        isAuthenticated: true,
        user: {
          id: 1,
        },
      },
    };
    nextProps = {
      loginUser: jest.fn(),
      checkoutOrder: jest.fn(),
      history: {
        push: jest.fn(),
      },
      success: {
        status: true,
      },
      orders: [],
      auth: {
        isAuthenticated: true,
        user: {
          id: 1,
        },
      },
    };
  });

  it('invokes `componentDidMount` when mounted', () => {
    const spy = jest.spyOn(LoginPage.prototype, 'componentDidMount');
    wrapper = shallow(<LoginPage {...props} />);
    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('invokes `componentWillRecieveProps` when mounted', () => {
    const spy = jest.spyOn(LoginPage.prototype, 'componentWillReceiveProps');
    wrapper = shallow(<LoginPage {...props} />);
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(spy).toHaveBeenCalled();
  });

  it('should render the component', () => {
    wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the Button component', () => {
    wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper.length).toBe(1);
  });
});
