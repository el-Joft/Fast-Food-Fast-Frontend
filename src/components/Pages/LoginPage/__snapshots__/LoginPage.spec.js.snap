// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
<section>
  <div
    className="error_message"
  />
  <div>
    <form
      onSubmit={[Function]}
    >
      <ul
        className="form-group"
      >
        <li>
          <label>
            Email
             
            <span
              className="required"
            >
              *
            </span>
          </label>
          <input
            className="form-control field-long"
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </li>
        <li>
          <label>
            Password
             
            <span
              className="required"
            >
              *
            </span>
          </label>
          <input
            className="form-control field-long"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </li>
        <li>
          <input
            type="submit"
            value="Submit"
          />
        </li>
      </ul>
    </form>
    <div
      className="text-form"
    >
      <p>
        Don't have an account?
         
        <Link
          replace={false}
          to="/signup"
        >
          Register
        </Link>
      </p>
    </div>
  </div>
</section>
`;
